lld/unittests/EVM/LLDTest.cpp

```diff
@@ -145,109 +158,360 @@ define i256 @foo() {                                              \n\
+  std::swap(OutMemBuf, InMemBuf);
+  LLVMDisposeMemoryBuffer(OutMemBuf);
 
-  // The first linker symbol definitions is provided, so we still have to
-  // receive two ELF object files, because of the undefined second reference.
-  if (LLVMLinkEVM(InMemBuf.data(), InIDs, 2, OutMemBuf.data(), LinkerSymbol,
-                  LinkerSymbolVal, 1, &ErrMsg)) {
+  // The first linker symbol definitions is provided, so we still have
+  // to receive an ELF object file
+  if (LLVMLinkEVM(InMemBuf, &OutMemBuf, LinkerSymbol, LinkerSymbolVal, 1,
+                  &ErrMsg)) {
     FAIL() << "Failed to link:" << ErrMsg;
     LLVMDisposeMessage(ErrMsg);
     return;
   }
 
-  EXPECT_TRUE(LLVMIsELFEVM(OutMemBuf[0]));
-  EXPECT_TRUE(LLVMIsELFEVM(OutMemBuf[1]));
-
-  LLVMGetUndefinedReferencesEVM(OutMemBuf[0], &UndefLinkerSyms,
-                                &NumLinkerUndefs);
-  EXPECT_TRUE(NumLinkerUndefs == 0);
-  LLVMDisposeUndefinedReferences(UndefLinkerSyms, NumLinkerUndefs);
+  EXPECT_TRUE(LLVMIsELFEVM(OutMemBuf));
 
-  LLVMGetUndefinedReferencesEVM(OutMemBuf[1], &UndefLinkerSyms,
-                                &NumLinkerUndefs);
+  LLVMGetUndefinedReferencesEVM(OutMemBuf, &UndefLinkerSyms, &NumLinkerUndefs);
+  EXPECT_TRUE(NumLinkerUndefs == 1);
   EXPECT_TRUE((std::strcmp(UndefLinkerSyms[0], LinkerSymbol[1]) == 0));
   LLVMDisposeUndefinedReferences(UndefLinkerSyms, NumLinkerUndefs);
 
-  InMemBuf.swap(OutMemBuf);
-  LLVMDisposeMemoryBuffer(OutMemBuf[0]);
-  LLVMDisposeMemoryBuffer(OutMemBuf[1]);
+  std::swap(OutMemBuf, InMemBuf);
+  LLVMDisposeMemoryBuffer(OutMemBuf);
 
   // Both linker symbol definitions are provided, so we have to receive
-  // bytecodes files.
-  if (LLVMLinkEVM(InMemBuf.data(), InIDs, 2, OutMemBuf.data(), LinkerSymbol,
-                  LinkerSymbolVal, 2, &ErrMsg)) {
+  // a bytecode.
+  if (LLVMLinkEVM(InMemBuf, &OutMemBuf, LinkerSymbol, LinkerSymbolVal, 2,
+                  &ErrMsg)) {
     FAIL() << "Failed to link:" << ErrMsg;
     LLVMDisposeMessage(ErrMsg);
     return;
   }
 
-  EXPECT_TRUE(!LLVMIsELFEVM(OutMemBuf[0]));
-  EXPECT_TRUE(!LLVMIsELFEVM(OutMemBuf[1]));
+  EXPECT_TRUE(!LLVMIsELFEVM(OutMemBuf));
 
-  LLVMGetUndefinedReferencesEVM(OutMemBuf[0], &UndefLinkerSyms,
-                                &NumLinkerUndefs);
+  LLVMGetUndefinedReferencesEVM(OutMemBuf, &UndefLinkerSyms, &NumLinkerUndefs);
   EXPECT_TRUE(NumLinkerUndefs == 0);
   LLVMDisposeUndefinedReferences(UndefLinkerSyms, NumLinkerUndefs);
 
-  LLVMGetUndefinedReferencesEVM(OutMemBuf[1], &UndefLinkerSyms,
-                                &NumLinkerUndefs);
+  StringRef DeployBin(LLVMGetBufferStart(OutMemBuf),
+                      LLVMGetBufferSize(OutMemBuf));
+
+  EXPECT_TRUE(DeployBin.find(SymVal1) != StringRef::npos);
+  EXPECT_TRUE(DeployBin.find(SymVal2) != StringRef::npos);
+
+  LLVMDisposeMemoryBuffer(OutMemBuf);
+  LLVMDisposeMemoryBuffer(InMemBuf);
+}
+
+TEST_F(LLDCTest, Assembly) {
+  StringRef A_deploy = "\
```

Comment:
Could we extract these IRs to *.ll files and put them to the Input directory?

Reply:
Done.